File: Generation/DSA01014.cpp
#include <bits/stdc++.h>
using namespace std;

int n, k, s, total, cnt;
vector<int> v;

void gen(int start) {
    if (total > s) return;
    if (v.size() > k) return;

    if (v.size() == k and total == s) {
        ++cnt;
        return;
    }
    
    for (int i = start; i <= n; ++i) {
        v.push_back(i);
        total += i;
        gen(i + 1);
        total -= i;
        v.pop_back();
    }
}

int main() {
    while (cin >> n >> k >> s, n || k || s) {
        v.clear();
        cnt = 0;
        total = 0;
        gen(1);

        cout << cnt << endl;
    }
    return 0;
}

File: Java/Generation/DSA01014.java
import java.util.Scanner;

public class DSA01014 {
    static int n, k, s, cnt;

    public static void gen(int start, int total, int count) {
        if (total > s || count > k) return; 
        if (count == k && total == s) {
            ++cnt;
            return;
        }

        for (int i = start; i <= n; ++i) {
            gen(i + 1, total + i, count + 1);
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        while (true) {
            n = scanner.nextInt();
            k = scanner.nextInt();
            s = scanner.nextInt();
            if (n == 0 && k == 0 && s == 0) break; 

            cnt = 0;
            gen(1, 0, 0);
            System.out.println(cnt);
        }

        scanner.close();
    }
}
